### DISCUSSION TITLE
Please explain __key in gatsby-source-filesystem

### DISCUSSION THREAD

**Discussion by don-esteban** [gatsbyjs/gatsby#33146] [Help]
Hi,

Could someone explain or link to an explanation of `__key` used in `gatsby-source-filesystem` plugin?

I checked this documents

- https://www.gatsbyjs.com/docs/reference/config-files/gatsby-config/
- https://www.gatsbyjs.com/plugins/gatsby-source-filesystem/

```
 {
  resolve: "gatsby-source-filesystem",
  options: {
    name: "images",
    path: "./src/images/",
  },
  __key: "images",
},
{
  resolve: "gatsby-source-filesystem",
  options: {
    name: "pages",
    path: "./src/pages/",
  },
  __key: "pages",
}
```
Thanks

**Comment by charsboo** [comment]
@don-esteban

I only see this property when a plugin is installed via Gatsby Recipes (the `gatsby new` command uses Gatsby Recipes).

According to the Gatsby Recipes [README](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-recipes/README.md#props), the `key` prop is a "string used to distinguish between multiple plugin instances."

**Comment by wardpeet** [comment]
Recipes uses it under the hood to make sure when changes happen it can still understand which plugin to edit. Think of it as a unique identifier for every plugin, if there are duplicates we still know which one to apply changes to. You can ignore it and even remove it after recipes is done.
